crates/cli/src/lib.rs

```diff
@@ -69,24 +69,53 @@ impl Args {
             Command::Start { config: config_path } => {
                 let config = CommitBoostConfig::from_file(&config_path);
 
+                // Initialize Docker client
+                let docker = bollard::Docker::connect_with_local_defaults().expect("Failed to connect to Docker");
+
                 if let Some(modules) = config.modules {
                     let signer_config = config.signer.expect("missing signer config with modules");
+                    // start signing server
+                    tokio::spawn(SigningService::run(config.chain, signer_config));
 
                     // this mocks the commit boost client starting containers, processes etc
                     let mut child_handles = Vec::with_capacity(modules.len());
 
                     for module in modules {
-                        let child = std::process::Command::new(module.path)
-                            .env(MODULE_ID_ENV, module.id)
-                            .env(CONFIG_PATH_ENV, &config_path)
-                            .spawn()
-                            .expect("failed to start process");
+                        match module.source {
+                            ModuleSource::DockerImageId(docker_image) => {
+                                let config = bollard::container::Config {
+                                    image: Some(docker_image.clone()),
+                                    host_config: Some(bollard::secret::HostConfig {
+                                        binds: {
+                                            let full_config_path = std::fs::canonicalize(&config_path).unwrap().to_string_lossy().to_string();
+                                            Some(vec![format!("{}:{}", full_config_path, "/config.toml")])
+                                        },
+                                        network_mode: Some(String::from("host")), // Use the host network
```

Comment:
It's worth noting here that I'm making substantial use of the [host network mode](https://docs.docker.com/network/drivers/host/) in order to share the networking namespace of the host with the module itself. 
This is sacrificing sandboxing for easier configuration. 

*NOTE* This is a [beta feature for Docker desktop](https://docs.docker.com/network/drivers/host/#docker-desktop) (if you're using that), and must be enabled explicitly.

Might not be the best long-term decision, but for simplicity's sake it's closer to native; just like the alternative (previous) approach of launching modules as subprocesses.